### PR TITLE
fix exception when ES6 class with no propTypes defined before calling requireNativeComponent

### PR DESCRIPTION
**Pull request by kejinlu**

### CODE REVIEW

**Comment by ide** [comment]
I have a question - what kind of native view does not have custom props? All the native views so far have special props to customize them.

File: Libraries/ReactIOS/verifyPropTypes.js [hunk 1]
```diff
@@ -24,7 +24,8 @@ function verifyPropTypes(
   }
   var nativeProps = viewConfig.nativeProps;
   for (var prop in nativeProps) {
-    if (!component.propTypes[prop] &&
+    if (component.propTypes != undefined &&
```
Comment (sahrens) [inline Libraries/ReactIOS/verifyPropTypes.js]: Please use `===` or `!==` instead of their looser counterparts whenever possible.

Comment (sahrens) [inline Libraries/ReactIOS/verifyPropTypes.js]: Or just check truthiness without comparing to `undefined`.

**Comment by vjeux** [comment]
Can you provide more context in how you experienced this error?

**Comment by kejinlu** [comment]
React Native "Native UI Components (iOS)" Guide 

call requireNativeComponent before  propTypes ，Then will experience this error

```
// MapView.js
var React = require('react-native');
var { requireNativeComponent } = React;

class MapView extends React.Component {
  render() {
    return <RCTMap {...this.props} />;
  }
}

var RCTMap = requireNativeComponent('RCTMap', MapView);

MapView.propTypes = {
  /**
   * When this property is set to `true` and a valid camera is associated
   * with the map, the camera’s pitch angle is used to tilt the plane
   * of the map. When this property is set to `false`, the camera’s pitch
   * angle is ignored and the map is always displayed as if the user
   * is looking straight down onto it.
   */
  pitchEnabled: React.PropTypes.bool,
};

module.exports = MapView;
```

**Comment by ide** [comment]
I think there's a bug in that code, since `pitchEnabled` will not get verified (docs fix at https://github.com/facebook/react-native/pull/1271). However it's also bad to get a confusing error about undefined not being an object, so I think we should change verifyPropTypes to assert that `component.propTypes` is defined and throw a helpful error message when it's missing.

**Comment by sahrens** [comment]
@ide is right - maybe change this PR to throw a more useful error?

**Comment by kejinlu** [comment]
Should I close the request?

**Comment by ide** [comment]
@kejinlu can you update this PR so that it asserts?

**Comment by kejinlu** [comment]
@ide  I found another problem ,the "component.displayName" in the verifyPropTypes  code always be undefined. Should set component.displayName with some value manually in the previous context

**Comment by sahrens** [comment]
> "component.displayName" in the verifyPropTypes code always be undefined

Looks like for ES6 classes `name` works, but for `React.createClass` `displayName` works.

@spicyj : any idea why this is inconsistent?  Is there a better mechanism than just checking both like

  '`' + (component.name || component.displayName) + '` has no propTypes defined`'

**Comment by sahrens** [comment]
import

**Comment by sahrens** [comment]
Can you figure out the right way to include the component name and make it consistent in the function?

**Comment by sahrens** [comment]
@facebook-github-bot import

**Comment by spicyj** [comment]
displayName || name is the right way to do it.

**Pull request closed without merging**